### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by filalex77**
The error in tests regarding `ssl_certificates_spec.rb` has been addressed at #17.

### CODE REVIEW

**Comment by filalex77** [comment]
The error in tests regarding `ssl_certificates_spec.rb` has been addressed at #17.

**Comment by taylorthurlow** [comment]
Just reviewing the changes, thanks for doing all of this. I'm not so sure on the Ruby version bump though. I had it at `2.2.10` because I wanted to provide some level of back support for older Ruby versions. That said, I know that `2.2` is basically unsupported, so if anything it should be a `2.3.x` release.

Perhaps it's the best idea to leave `.ruby-version` at `2.5.1` and just let TravisCI alert when we make a breaking change for `2.4.x` or `2.3.x`. Thoughts?

**Comment by filalex77** [comment]
I think you're right. `.ruby-version` should be the latest supported one, not the oldest supported. Travis should check all major versions starting with 2.2, so we don't break anything. Good thinking!

**Comment by taylorthurlow** [comment]
I like all of your changes, but builds fail because `delete_suffix` wasn't implemented until Ruby `2.5`. We should switch to something compatible with `2.3` onwards.

**Comment by filalex77** [comment]
Ah, I didn't see that, sorry. I think we can go with `String#sub` then.

**Comment by taylorthurlow** [comment]
Running into some issues with this merge, so I hard reset the develop branch back to where it was before the merge. Trying to sort out the fact that there are a bunch of these old merge commits attached to this PR - it creates a bunch of merge bubbles that I don't want in git history. Going to try to rebase the relevant changes onto the top of develop.

File: .rubocop.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 require: rubocop-rspec
 
 AllCops:
-  TargetRubyVersion: 2.2
+  TargetRubyVersion: 2.5.1
```
[the remark this answers is not in the document]
Comment (filalex77) [reply .rubocop.yml]: Sure, that is a good idea. 2.3+ compat is perfectly fine for me.

**Comment by taylorthurlow** [comment]
I moved develop back to the point before you made any changes, and then rebased your changes onto develop. This seems to have done the trick. I did this all in the refactor branch then merged it.

Not exactly sure how to avoid this problem in the future - is it possible that this was caused by merging the develop changes into your refactor branch?

**Comment by filalex77** [comment]
@taylorthurlow It probably was. Though I'm not exactly sure too :)

**Pull request merged**